files/check_zpool_scrub

```diff
@@ -0,0 +1,527 @@
+# they are used in some global variables.
+
+# to_year ###
+
+##
+# Get the four digit year integer from now.
+#
+# Return:
+#   The current 4 digit year.
+##
+now_to_year() {
+	date +%Y
+}
+
+##
+# Convert a date in the format YYYY-MM-DD to a four digit year integer.
+#
+# Parameters:
+#   a date in the format YYYY-MM-DD
+#
+# Return:
+#   four digit year integer
+##
+date_to_year() {
+	date --date $1 +%Y
+}
+
+# to_datetime ###
+
+##
+# Convert a UNIX timestamp to a datetime string.
+#
+# Parameters:
+#   UNIX timestamp
+#
+# Return:
+#   %Y-%m-%d.%H:%M:%S
+##
+timestamp_to_datetime() {
+	date --date @$1 +%Y-%m-%d.%H:%M:%S
+}
+
+# to_timestamp ###
+
+##
+# Get the current UNIX timestamp.
+#
+# Return:
+#   %current UNIX timestamp
+##
+now_to_timestamp() {
+	date +%s
+}
+
+##
+# Convert a datetime in the ctime format to a Unix timestamp.
+# ctime is a textual representation of a datetime derived from the
+# c function ctime (https://en.cppreference.com/w/c/chrono/ctime).
+# The ctime string has the following format:  Www Mmm dd hh:mm:ss yyyy,
+# where Www is the weekday, Mmm the month in letters, dd the day of the
+# month, hh:mm:ss the time, and yyyy the year.
+#
+# see https://www.freebsd.org/cgi/man.cgi?query=strftime&sektion=3
+#  %c is replaced by national representation of time and date.
+#
+# Parameters:
+#   $1: Www Mmm dd hh:mm:ss yyyy
+#
+# Return:
+#   UNIX timestamp
+##
+ctime_to_timestamp() {
+	date --date "$1" +%s
+}
+
+########################################################################
+# Global variables
+########################################################################
+
+. "$(dirname "${0}")/common.sh"
+
+checkName=zpool-scrub
+checkOutput=""
+
+USAGE="
+Usage: $0 <options>
+
+Options:
+ -c, --critical=OPT_CRITICAL
+    Interval in seconds for critical state.
+ -w, --warning=OPT_WARNING
+    Interval in seconds for warning state. Must be lower than -c.
+ -p,--pool=OPT_POOL
+    Name of the pool. If this option is omitted all pools are checked.
+ -h, --help
+    Show this help.
+
+Performance data:
+
+POOL is the name of the pool
+
+ - POOL_last_ago
+    Time interval in seconds for last scrub.
+ - POOLprogress
+    Percent 0 - 100
+ - POOLspeed
+    MB per second.
+ - POOL_time
+    Time to go in minutes.
+
+Details about the implementation of this monitoring plugin:
+
+This monitoring plugin grabs the last scrub date from the command
+'zpool status POOL'.
+"
+
+# Options from the command line interface
+OPT_POOL=
+# 1 month 60*60*24*31
+OPT_WARNING=2678400
+# 2 month 60*60*24*31*2
+OPT_CRITICAL=5356800
+
+_getopts() {
+	local OPT
+	while getopts ':c:hp:svw:-:' OPT ; do
+		case $OPT in
+			c) OPT_CRITICAL=$OPTARG ;;
+			h) echo "$USAGE" ; exit 0 ;;
+			p) OPT_POOL="$OPTARG" ;;
+			w) OPT_WARNING=$OPTARG ;;
+
+			\?)
+				echo "Invalid option “-$OPTARG”!" >&2
+				exit 2
+				;;
+
+			:)
+				echo "Option “-$OPTARG” requires an argument!" >&2
+				exit 3
+				;;
+
+			-)
```

Comment:
Are you sure we need that? We could only support short parameters and lower code complexity

Reply:
No. I actually don't want to touch that part because I hate bash and am not very familiar with the syntax.